src/main/antora/modules/security/pages/security.adoc

```diff
@@ -1713,4 +1713,222 @@ The test itself is basically the same as that for the <<Securing an endpoint wit
+. How to define (and implicitly set) a custom https://jakarta.ee/specifications/security/3.0/apidocs/jakarta.security/jakarta/security/enterprise/identitystore/package-summary.html[identity store,role=external,window=_blank]
+include::partial$list-item-learn-security-context.adoc[]
+
+=== Write the application
+
+include::partial$rest-resource-class.adoc[]
+
+==== Declare the security constraints
+
+include::partial$declare-security-constraints.adoc[]
+
+==== Declare the authentication mechanism
+
+[source,java]
+----
+@ApplicationScoped
+@CustomFormAuthenticationMechanismDefinition(
+    loginToContinue = @LoginToContinue(
+        loginPage="/login.xhtml",
+        errorPage=""
+    )
+)
+@DeclareRoles({ "user", "caller" })
+@FacesConfig
+@ApplicationPath("/rest")
+public class ApplicationConfig extends Application {
+
+}
+----
+
+include::partial$declare-authentication-mechanism.adoc[]
+
+Contrary to the Basic HTTP authentication mechanism, the Form authentication mechanism allows us to customize the login dialog and to keep track of the authenticated session on the server (using a cookie). This also allows us to logout, something that for unknown reasons has never been specified for Basic HTTP authentication.
+
+Contrary to the regular Form authentication mechanism, the Custom Form authentication mechanism let's us customize the login dialog even more by having the ability to execute custom code between the postback of a login form and the form authentication mechanism taking the provided credentials.
+
+To use this authentication method, we need to designate a path to a resource that is relative to our application. The authentication mechanism will redirect the caller to this resource when authentication is required. The resource can be anything, but a postback should eventually lead to some code being executed that continues the authentication dialog. E.g. a plain .html file or .jsp file combined with  a Filter, or a Faces view with a backing bean.
+
+
+==== Define the authentication mechanism's view and backing code
+
+For this example we'll use a Faces view with a backing bean:
+
+[source,xml]
+----
+<html lang="en"
+    xmlns="http://www.w3.org/1999/xhtml"
+    xmlns:h="jakarta.faces.html"
+    xmlns:faces="jakarta.faces"
+>
+    <h:head/>
+    <h:messages />
+    
+    <body>
+        <p>
+            Login to continue
+        </p>
+    
+         <form faces:id="form">
+            <p>
+                <strong>Username </strong> 
+                <input faces:id="username" type="text" faces:value="#{loginBacking.username}" />
+            </p>
+            <p>
+                <strong>Password </strong> 
+                <input faces:id="password" type="password" faces:value="#{loginBacking.password}" />
+            </p>
+            <p>
+                <input type="submit" value="Login" faces:action="#{loginBacking.login}" />
+            </p>
+        </form>
```

Comment:
I'm wondering the rationale to use passthrough elements.

And these strongs should be <label for> or <h:outputLabel>.

Reply:
As I remember, it was to be as close as possible to the .html version used in the previous example.